Sources/TuistGraph/Models/Target.swift

```diff
@@ -83,7 +83,13 @@ public struct Target: Equatable, Hashable, Comparable, Codable {
         self.product = product
         self.destinations = destinations
         self.bundleId = bundleId
-        self.productName = productName ?? name.sanitizedModuleName
+        if let productName {
+            self.productName = productName
+        } else if product == .framework || product == .staticFramework {
```

Comment:
is the discrepancy of what constitutes of valid product names documented anywhere? I wasn't aware frameworks an applications have different rules 🤔

Reply:
It's the result of my manual research. I created an empty project in Xcode and created all possible types of targets with names containing dots and hyphens. `Framework` is the only target where Xcode transforms Product Name by setting it to "$(TARGET_NAME:c99extidentifier)". Product Name for all other targets was just "$(TARGET_NAME)"

Reply:
I would suggest to try to bring the same approach in Tuist someday. But, taking into account your previous comment, not in this PR.